src/Cli/func/Common/Utilities.cs

```diff
@@ -293,5 +293,46 @@ internal static bool IsMinifiedVersion()
 
             return false;
         }
+
+        internal static string ExtractIKeyFromConnectionString(string connectionString)
+        {
+            if (string.IsNullOrEmpty(connectionString))
+            {
+                return null;
+            }
+
+            var span = connectionString.AsSpan();
+            var start = 0;
+
+            while (start < span.Length)
+            {
+                var semicolonIndex = span.Slice(start).IndexOf(';');
```

Comment:
Is this the only way to extract IKey from connection string? Do we not have any builtin/standard method for this?

Reply:
nit: alternative approach -

```
    return input?
        .Split(';')
        .Select(s => s.Split('='))
        .Where(parts => parts.Length == 2 && parts[0].Trim().Equals("InstrumentationKey", StringComparison.OrdinalIgnoreCase))
        .Select(parts => parts[1].Trim())
        .FirstOrDefault();
```